msal/src/main/java/com/microsoft/identity/client/IPublicClientApplication.java

```diff
@@ -84,6 +85,19 @@ void acquireToken(@NonNull final Activity activity,
     @WorkerThread
     IAuthenticationResult acquireTokenSilent(@NonNull final AcquireTokenSilentParameters acquireTokenSilentParameters) throws InterruptedException, MsalException;
 
+    /**
+     * Perform the Device Code Flow (DCF) protocol to allow a deviceto authenticate and get a new access token without input capability.
+     * @param scopes
+     * @param callback The {@link DeviceCodeFlowCallback} object to receive responses from the protocol.
+     *                 1). Receiving authentication information (user_code, verification_uri, and instruction message)
+     *                 via {@link DeviceCodeFlowCallback#getUserCode(String, String, String)}.
+     *                 2). Receiving a successful authnetication result containing a fresh access token
+     *                 via {@link DeviceCodeFlowCallback#getToken(AuthenticationResult)}.
+     *                 3). Receiving an exception detailing what went wrong in the protocol
+     *                 via {@link DeviceCodeFlowCallback#onError(MsalException)}.
+     */
```

Comment:
We may not need to document all this here, and rather it would be better to document these in the javadoc for `DeviceCodeFlowCallback` class and over here a simple one line definition of a callback should suffice